app/models/person.rb

```diff
@@ -104,28 +104,38 @@ def my_delegated_protocol_subscriptions
     ProtocolSubscription.active.where(filling_out_for_id: id).where.not(person_id: id)
   end
 
+  # For any method that only returns open responses, we want them to be sorted by descending priority first,
+  # and ascending open_from second. This is because we call .first on this method to determine which is the next
+  # response that should be shown to the user.
   def my_open_responses(for_myself = true)
     active_subscriptions = protocol_subscriptions.active if for_myself.blank?
     active_subscriptions ||= my_protocols(for_myself)
-    active_subscriptions.map { |prot| prot.responses.opened_and_not_expired }.flatten.sort_by(&:open_from)
+    active_subscriptions.map { |prot| prot.responses.opened_and_not_expired }.flatten.sort_by(&:priority_sorting_metric)
```

Comment:
don't you need to add sorting for the open from within the priority queue too? so, somehow, sort by priority, and then (taking that priority into account) sort by open_from?

Reply:
right, nvm, you did that in the next part.

Reply:
Yep. The priority sorting metric just means priority first, open from second.